hera_mc/mc_session.py

```diff
@@ -1396,6 +1398,129 @@ def get_node_power_command(self, most_recent=None, starttime=None, stoptime=None
                                  starttime=starttime, stoptime=stoptime,
                                  filter_column='node', filter_value=nodeID)
 
+    def add_correlator_config(self, time, config_hash, config_file):
+        """
+        Add new correlator config to the M&C database.
+
+        Parameters:
+        ------------
+        time: astropy time object
+            astropy time object based on a timestamp reported by the correlator
+        config_hash: string
+            unique hash of the config
+        config_file: string
+            name of the config file in the Librarian
+        """
+        from .correlator import CorrelatorConfig
+
+        self.add(CorrelatorConfig.create(time, config_hash, config_file))
+
+    def add_correlator_config_from_corrcm(self, config_state_dict=None,
+                                          testing=False):
+        """Get and add correlator config information using a HeraCorrCM object.
+
+        This function connects to the correlator and gets the latest config using
+        the `corr._get_config` function. If it is a new config file, it connects
+        to the Librarian to upload the config file. For testing purposes, it can
+        optionally accept an input dict instead of connecting to the correlator.
+
+
+        Parameters:
+        ------------
+        config_state_dict: dict
+            A dict containing info as in the return dict from _get_config() for
+            testing purposes. If None, _get_config() is called. Default: None
+        testing: boolean
+            If true, don't add config file to the Librarian or add a record of
+            it to the database and return the CorrelatorConfig object. Default False.
+
+        Returns:
+        --------
+        Optionally returns the CorrelatorConfig (if testing is True)
+        """
+        from .correlator import _get_config, CorrelatorConfig
+
+        if config_state_dict is None:
+            config_state_dict = _get_config()
+
+        time = Time(config_state_dict['timestamp'], format='unix')
+        config = config_state_dict['config']
+        config_hash = config_state_dict['hash']
+
+        # first check to see if a file with this hash (so identical) already exists
+        same_config = self.get_correlator_config(most_recent=True, config_hash=config_hash)
+        if len(same_config) == 0:
+            config_filename = 'correlator_config_' + str(int(floor(time.gps))) + '.yaml'
+            config_obj = CorrelatorConfig.create(time, config_hash, config_filename)
+
+            if not testing:  # pragma: no cover
+                # This config is new. Need to save it to the Librarian
```

Comment:
@pkgw can you read this section of this method? It's for adding config files to the librarian and I want to make sure it's right.